Game_Development/Dodge-the-Blocks/main.py

```diff
@@ -0,0 +1,314 @@
+    button_color = GREEN
+    button_rect = pygame.Rect(0, 0, 300, 70)
+    button_rect.center = (screen_rect.centerx, screen_rect.centery + 200)
+    
+    pygame.draw.rect(screen, button_color, button_rect)
+    pygame.draw.rect(screen, BLACK, button_rect, 3)  # Add border
+    
+    start_text = font.render("START", True, BLACK)
+    start_text_rect = start_text.get_rect(center=button_rect.center)
+    screen.blit(start_text, start_text_rect)
+    
+    pygame.display.flip()
+
+    # Wait for user to click the Start button
+    waiting = True
+    while waiting:
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                pygame.quit()
+                quit()
+            elif event.type == pygame.MOUSEBUTTONDOWN:
+                mouse_pos = pygame.mouse.get_pos()
+                if button_rect.collidepoint(mouse_pos):
+                    waiting = False  # Exit the loop and start the game
+
+        # Change the cursor to a pointer when hovering over the button
+        mouse_pos = pygame.mouse.get_pos()
+        if button_rect.collidepoint(mouse_pos):
+            pygame.mouse.set_cursor(pygame.SYSTEM_CURSOR_HAND)  # Set cursor to pointer
+        else:
+            pygame.mouse.set_cursor(pygame.SYSTEM_CURSOR_ARROW)  # Set cursor to default arrow
+
+
+# General utility functions:
+def handle_player_movement(event, player):
+    if event.type == pygame.KEYDOWN:
+        if event.key == pygame.K_LEFT:
+            player.rect.x -= 3
+        if event.key == pygame.K_RIGHT:
+            player.rect.x += 3
+        if event.key == pygame.K_UP:
+            player.rect.y -= 3
+        if event.key == pygame.K_DOWN:
+            player.rect.y += 3
+
+def display_score(screen, score, wall_touch_count, level):
+    font = pygame.font.SysFont('comicsansms', 30)
+    score_text = font.render(f"Score: {score - wall_touch_count}", True, DARK_BLUE)
+    level_text = font.render(f"Level: {level}", True, DARK_BLUE)
+    screen.blit(score_text, [10, 10])
+    screen.blit(level_text, [10, 50])
+
+def move_blocks(block, index, l1, l2):
+    block.rect.x += l1[index]
+    block.rect.y += l2[index]
+    if block.rect.x >= 840 or block.rect.x < 10:
+        l1[index] *= -1
+    if block.rect.y >= 540 or block.rect.y < 10:
+        l2[index] *= -1
+
+
+# Block class definition
+class Block(pygame.sprite.Sprite):
+    
+    def __init__(self, color, width, height):
+        
+        super().__init__()
+        
+        self.image = pygame.Surface([width, height])
+        self.image.fill(color)
+ 
+        self.rect = self.image.get_rect()
+
+
+# Main game loop and logic
+pygame.init()
+pygame.mixer.init()
+size = (900,600)
+screen = pygame.display.set_mode((size))
+pygame.display.set_caption("My Game")
+
+pygame.mixer.music.load(welcome_screen_music)
+start_screen(screen)
+pygame.mixer.music.stop()
+pygame.mixer.music.load(in_game_music)
+pygame.mixer.music.play(-1)
+
+# Main game loop and logic
+pygame.init()

```

Comment:
Don't need this twice. Initialisation required just at the start.

Reply:
Removed second pygame.init(). Thanks for the correction